### ISSUE TITLE
Inconsistency when building weapon sets for melee attacks

### ISSUE THREAD

**Issue by IceRaptor** [BattletechModders/CleverGirl#5]
Reported by Harkonnen in closed-3437 on RT discord:

```
1. It seems that whenever weapon fires in melee is hardcoded to Antipersonall
2020-04-09 01:11:53.876 -  (Weapon_Laser_LargeLaserPulse_CLAN) is out of range (MaxRange: 840 vs 184.4063) or has no LOF, skipping.
2020-04-09 01:11:53.876 -  (Weapon_SRM_SRM_NARC_CLAN) is out of range (MaxRange: 480 vs 184.4063) or has no LOF, skipping.
2020-04-09 01:11:53.876 -  (Weapon_SRM_SRM_NARC_CLAN) is anti-personnel, adding to melee and DFA sets.
2020-04-09 01:11:53.876 -  (Weapon_Laser_Light_TAG_CLAN) is out of range (MaxRange: 510 vs 184.4063) or has no LOF, skipping.
2020-04-09 01:11:53.876 -  (Weapon_Laser_Light_TAG_CLAN) is anti-personnel, adding to melee and DFA sets.
2020-04-09 01:11:53.876 -  (Weapon_Elite_2_Chainsword) is out of range (MaxRange: 40 vs 184.4063) or has no LOF, skipping.
It added NARC nad TAG which dont fire in melle into melee sets, but didnt added Weapon_Elite_2_Chainsword which is a melee firing weapon
```

**Comment by IceRaptor** [comment]
Core issue may be types associated with anti-personnel aren't making sense in the RT/BTA world.

**Comment by IceRaptor** [comment]
[clever_girl.log.1.zip](https://github.com/IceRaptor/CleverGirl/files/4475985/clever_girl.log.1.zip)

**Comment by IceRaptor** [comment]
Should be fixed in 0.4.5
